build/helper.py

```diff
@@ -136,23 +143,38 @@ def add_all_metadata(functions):
             _add_buffer_info(p)
     return functions
 
-# Normalize string type between python2 & python3
+# Python 2/3 compatibility
+
 def normalize_string_type(d):
+    '''Normalize string type between python2 & python3'''
     if sys.version_info.major < 3:
         if type(d) is dict:
             for k in d:
                 d[k] = normalize_string_type(d[k])
         elif type(d) is str:
             d = d.decode('utf-8')
-
     return d
+
 # Functions that return snippets that can be placed directly in the templates.
 
-def get_method_parameters_snippet(parameters):
-    '''Returns a string suitable for the parameter list of a method given a list of parameter objects'''
+def get_method_parameters_snippet(parameters, skip_session_handle, skip_ivi_dance_size_parameter, skip_output_parameters):
```

Comment:
This is one of the more interesting changes I think. Instead of caller having to filter out unwanted parameters, now caller can pass all function parameters and specify what to skip. What do you think?